src/jdk.incubator.jextract/share/classes/jdk/internal/clang/LibClang.java

```diff
@@ -71,16 +77,24 @@ public static Index createIndex(boolean local) {
         return index;
     }
 
-    public static String CXStrToString(MemorySegment cxstr) {
+    public static String CXStrToString(Function<SegmentAllocator, MemorySegment> segmentSupplier) {
+        MemorySegment cxstr = segmentSupplier.apply(stringAllocator);
```

Comment:
I don't see `stringAllocator` anywhere, I guess this should be `STRING_ALLOCATOR`.
```suggestion
        MemorySegment cxstr = segmentSupplier.apply(STRING_ALLOCATOR);
```